lib/elixir/lib/kernel.ex

```diff
@@ -2055,7 +2055,8 @@ defmodule Kernel do
   """
   @spec inspect(Inspect.t(), keyword) :: String.t()
   def inspect(term, opts \\ []) when is_list(opts) do
-    opts = struct(Inspect.Opts, opts)
+    custom_options = Inspect.Opts.filter_custom_options(opts)
```

Comment:
I would like the custom options to be given nested:

    custom_options: [...]

Otherwise, this basically makes it impossible for Elixir to add any new key in the future, as it may now conflict with a a user key.

Reply:
I thought a lot about this and I finally came to the conclusion that user’s convenience is more valuable since Elixir API was kinda declared nearly frozen.

Anyways, fixed.